cmake/FindSanitizers.cmake

```diff
@@ -20,54 +20,63 @@
 # Copyright (C) 2018 Scylladb, Ltd.
 #
 
-include (CheckCXXSourceCompiles)
+if(NOT Sanitizers_FIND_COMPONENTS)
+  set(Sanitizers_FIND_COMPONENTS
+    address
+    undefined_behavior)
+endif()
 
-set (CMAKE_REQUIRED_FLAGS -fsanitize=address)
-check_cxx_source_compiles ("int main() {}" Sanitizers_ADDRESS_FOUND)
-
-if (Sanitizers_ADDRESS_FOUND)
-  set (Sanitizers_ADDRESS_COMPILER_OPTIONS -fsanitize=address)
-endif ()
+foreach (component ${Sanitizers_FIND_COMPONENTS})
+  string (TOUPPER ${component} COMPONENT)
+  set (compile_options "Sanitizers_${COMPONENT}_COMPILE_OPTIONS")
+  if (component STREQUAL "address")
+    list (APPEND ${compile_options} -fsanitize=address)
+  elseif (component STREQUAL "undefined_behavior")
+    list (APPEND ${compile_options} -fsanitize=undefined)
+    # Disable vptr because of https://gcc.gnu.org/bugzilla/show_bug.cgi?id=88684
+    list (APPEND ${compile_options} -fno-sanitize=vptr)
```

Comment:
Note: this was fixed in gcc 10 (or perhaps 9?) - https://github.com/gcc-mirror/gcc/commit/c24847a54abe6b93139f92318f254d1952315ba2 so we can drop it later.

Reply:
sure. will do in a separate PR.

Reply:
created #2430